terminus/theme.py

```diff
@@ -147,6 +183,14 @@ def plugin_loaded():
         lambda _: sublime.active_window().run_command("terminus_generate_theme")
     )
 
+    def check_update_theme(value):
+        settings = sublime.load_settings("Terminus.sublime-settings")
+        if settings.get("theme", "default") == "adaptive":
+            sublime.active_window().run_command("terminus_generate_theme")
+
+    settings_on_change(sublime.load_settings("Preferences.sublime-settings"), "color_scheme")(
+        check_update_theme
+    )
 
 def plugin_unloaded():
     settings = sublime.load_settings("Terminus.sublime-settings")
```

Comment:
we need 
```
settings_on_change(sublime.load_settings("Preferences.sublime-settings"), "color_scheme", clear=True)
```
here.

By the way, I spotted a bug in `settings_on_change`, you might want to rebase to the master branch.

Reply:
I rebased on master, but if I use `clear=True`, settings_on_change will return `None` instead of a function. It seems to work without `clear` though.

Reply:
Ah sorry, you meant in `plugin_unloaded`